vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -8,85 +8,126 @@
+import org.springframework.core.io.ByteArrayResource;
+import org.springframework.core.io.Resource;
 import org.springframework.stereotype.Component;
 
 import edu.asu.diging.vspace.core.exception.FileStorageException;
 import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.model.IVSFile;
 import edu.asu.diging.vspace.core.model.IVSImage;
 
 @Component
 @PropertySource({"classpath:config.properties", "${appConfigFile:classpath:}/app.properties"})
 public class StorageEngine implements IStorageEngine {
 
-	@Value("${uploads_path}")
-	private String path;
-	
-	/* (non-Javadoc)
-	 * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
-	 */
-	@Override
-	public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
-		File parent = new File(path + File.separator + directory);
-		if (!parent.exists()) {
-			parent.mkdir();
-		}
-		File file = new File(parent.getAbsolutePath() + File.separator + filename);
+    @Value("${uploads_path}")
+    private String path;
+
+    @Value("${file_uploads_directory}")
+    private String fileUploadDir;
+
+    /* (non-Javadoc)
+    * @see edu.asu.diging.vspace.core.file.impl.IStorageEngine#storeFile(byte[], java.lang.String, java.lang.String)
+    */
+    @Override
+    public String storeFile(byte[] fileContent, String filename, String directory) throws FileStorageException {
+        File parent = new File(path + File.separator + directory);
+	if (!parent.exists()) {
+            parent.mkdir();
+        }
+        File file = new File(parent.getAbsolutePath() + File.separator + filename);
         BufferedOutputStream stream;
         try {
             stream = new BufferedOutputStream(new FileOutputStream(file));
         } catch (FileNotFoundException e) {
-            throw new FileStorageException("Could not store file.", e);
+	    throw new FileStorageException("Could not store file.", e);
         }
         try {
             stream.write(fileContent);
             stream.close();
         } catch (IOException e) {
             throw new FileStorageException("Could not store file.", e);
         }
-        
+
         return directory;
-	}
+    }
+
+    @Override
+    public byte[] getImageContent(String directory, String filename) throws IOException {
+        File fileObject = getFile(directory, filename);       
+        return getFileContent(fileObject);
+    }
 	
-	@Override
-	public byte[] getImageContent(String directory, String filename) throws IOException {
-		File fileObject = new File(path + File.separator + directory + File.separator + filename);
+    /**
+    * Method to rename image   
+    * 
+    * @param image - image file
+    * @param newFileName - new name of the file
+    * @return true if file renaming was successful, otherwise return false 
+    */ 
+    @Override
+    public boolean renameImage(IVSImage image, String newFileName) {
```

Comment:
`IVSImage` and `IVSFile` should have a common super interface (or maybe `IVSImage` should be a subtype of `IVSFile`?). Then you need only one method here.

Reply:
and actually, this class should just get a string anyways (it shouldn't need to know about our domain classes). All it should know about are how to get stuff from the file system and store stuff, no matter where the file to be stored comes from (image or file).